### DISCUSSION TITLE
Notification beforecreate and stop / return

### DISCUSSION THREAD

**Discussion by jcgmalave** [filamentphp/filament#4302] [Help] [answered]
Is it possible when using the beforecreate or before save function, to perform a validation and if it is fulfilled, send a notification and stop the execution of the create or save until it is corrected? and return to the main resource

**Comment by danharrin** [comment]
Send the notification, throw a ValidationException

**Reply by Z3d0X** [reply]
This could be considered a breaking change, but are you open to suggestion of stopping execution in "before" hooks by returning false?

I know most devs who asked this question so far are trying to do some sort of validation (which can be handled by a custom rule or, like you mentioned, by throwing ValidationException). However what about the potential use case of checking condition unrelated to the form?

**Reply by danharrin** [reply]
Feature request - #4380

**Comment by danharrin** [comment, marked as answer]
Implemented in #4433
